homeassistant/components/aquacell/sensor.py

```diff
@@ -0,0 +1,157 @@
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfTime
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AquacellCoordinator
+from .entity import AquacellEntity
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class SoftenerEntityDescription(SensorEntityDescription):
+    """Describes Softener sensor entity."""
+
+    value_fn: Callable[[Softener], str | int | float | None]
+
+
+SENSORS: tuple[SoftenerEntityDescription, ...] = (
+    SoftenerEntityDescription(
+        key="salt_leftpercentage",
+        translation_key="salt_leftpercentage",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:basket-fill",
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda softener: cast(int, softener.salt.leftPercent),
+    ),
+    SoftenerEntityDescription(
+        key="salt_rightpercentage",
+        translation_key="salt_rightpercentage",
+        icon="mdi:basket-fill",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda softener: cast(int, softener.salt.rightPercent),
+    ),
+    SoftenerEntityDescription(
+        key="salt_leftdays",
+        translation_key="salt_leftdays",
+        icon="mdi:calendar-today",
+        native_unit_of_measurement=UnitOfTime.DAYS,
+        value_fn=lambda softener: softener.salt.leftDays,
+    ),
+    SoftenerEntityDescription(
+        key="salt_rightdays",
+        translation_key="salt_rightdays",
+        icon="mdi:calendar-today",
+        native_unit_of_measurement=UnitOfTime.DAYS,
+        value_fn=lambda softener: softener.salt.rightDays,
+    ),
+    SoftenerEntityDescription(
+        key="fw_version",
+        translation_key="fw_version",
+        icon="mdi:chip",
+        value_fn=lambda softener: softener.fwVersion,
+    ),
+    SoftenerEntityDescription(
+        key="name",
+        translation_key="name",
+        value_fn=lambda softener: softener.name,
+    ),
+    SoftenerEntityDescription(
+        key="battery",
+        translation_key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda softener: softener.battery,
+    ),
+    SoftenerEntityDescription(
+        key="last_update",
+        translation_key="last_update",
+        icon="mdi:update",
+        value_fn=lambda softener: datetime.fromtimestamp(
+            cast(float, softener.lastUpdate) / 1000
+        ).isoformat(),
+    ),
+    SoftenerEntityDescription(
+        key="wifi_level",
+        translation_key="wifi_level",
+        icon="mdi:wifi",
+        value_fn=lambda softener: softener.wifiLevel,
+    ),
```

Comment:
What are the possible states?

Reply:
It's reporting in a string type, for example currently mine is 'high'. I haven't mapped out all possible states yet, probably something like high, mid, low. I'll add icon translations for that in a next PR.

Reply:
We should make this an enum sensor which can hold the possible values and allow translatable states.